cloud.ks

```diff
@@ -11,118 +10,98 @@ rootpw --lock --iscrypted locked
-cat > /etc/sysconfig/network << EOF
+cat <<EOF > /etc/sysconfig/network
 NETWORKING=yes
 NOZEROCONF=yes
 EOF
 
-# Remove any persistent NIC rules generated by udev
-rm -vf /etc/udev/rules.d/*persistent-net*.rules
-# And ensure that we will do DHCP on eth0 on startup
-cat > /etc/sysconfig/network-scripts/ifcfg-eth0 << EOF
+# simple eth0 config, again not hard-coded to the build hardware
+cat <<EOF > /etc/sysconfig/network-scripts/ifcfg-eth0
 DEVICE="eth0"
 BOOTPROTO="dhcp"
 ONBOOT="yes"
 TYPE="Ethernet"
 PERSISTENT_DHCLIENT="yes"
+NM_CONTROLLED="yes"
 EOF
 
+# generic localhost names
+cat <<EOF > /etc/hosts
+127.0.0.1   localhost localhost.localdomain localhost4 localhost4.localdomain4
+::1         localhost localhost.localdomain localhost6 localhost6.localdomain6
+
+EOF
+
+
 # Because memory is scarce resource in most cloud/virt environments,
 # and because this impedes forensics, we are differing from the Fedora
 # default of having /tmp on tmpfs.
-echo "Disabling tmpfs for /tmp."
 systemctl mask tmp.mount
 
-# Uncomment this if you want to use cloud init but suppress the creation
-# of an "ec2-user" account. This will, in the absence of further config,
-# cause the ssh key from a metadata source to be put in the root account.
-#cat <<EOF > /etc/cloud/cloud.cfg.d/50_suppress_ec2-user_use_root.cfg
-#users: []
-#disable_root: 0
-#EOF
+# make sure firstboot doesn't start
+echo "RUN_FIRSTBOOT=NO" > /etc/sysconfig/firstboot
 
-echo "Removing random-seed so it's not the same in every image."
-rm -f /var/lib/systemd/random-seed
-
-echo "Packages within this cloud image:"
-echo "-----------------------------------------------------------------------"
-rpm -qa
-echo "-----------------------------------------------------------------------"
-# Note that running rpm recreates the rpm db files which aren't needed/wanted
-rm -f /var/lib/rpm/__db*
-
-echo "Zeroing out empty space."
-# This forces the filesystem to reclaim space from deleted files
-dd bs=1M if=/dev/zero of=/var/tmp/zeros || :
-rm -f /var/tmp/zeros
-echo "(Don't worry -- that out-of-space error was expected.)"
-
-# For trac ticket https://pagure.io/atomic-wg/issue/128
-rm -f /etc/sysconfig/network-scripts/ifcfg-ens3
-
-# Disable network service here, as doing it in the services line
-# fails due to RHBZ #1369794
-/sbin/chkconfig network off
-
-# Anaconda is writing an /etc/resolv.conf from the install environment.
-# The system should start out with an empty file, otherwise cloud-init
-# will try to use this information and may error:
-# https://bugs.launchpad.net/cloud-init/+bug/1670052
+# Additional drivers for vmware and hyperv
+echo 'add_drivers+="mptspi vmw_pvscsi "'    > /etc/dracut.conf.d/vmware-fusion-drivers.conf
+echo 'add_drivers+="hv_storvsc hv_netvsc "' > /etc/dracut.conf.d/hyperv-drivers.conf
+# Rerun dracut for the installed kernel (not the running kernel):
+KERNEL_VERSION=$(rpm -q kernel --qf '%{V}-%{R}.%{arch}\n')
+dracut -f /boot/initramfs-$KERNEL_VERSION.img $KERNEL_VERSION
```

Comment:
This whole section never worked...but I guess we should fix that upstream first.